### PR TITLE
fix: don't log severe when an asset URI cannot be resolved

### PR DESCRIPTION
**Pull request by kevmoo**
Follow-up to https://github.com/dart-lang/json_serializable/issues/236

### CODE REVIEW

**Comment by kevmoo** [comment]
So throwing for dart: URIs lets us succeed, but `log.severe` fails.

Weird?

**Comment by natebosch** [comment]
>So throwing for dart: URIs lets us succeed, but `log.severe` fails.
> Weird?

If the builders catches the exception this is expected. We always fail for `log.severe` but we only fail for exceptions the build system catches, not those that the builder catches and swallows.

File: build_resolvers/lib/src/resolver.dart [hunk 1]
```diff
@@ -289,17 +289,21 @@ class _AssetUriResolver implements UriResolver {
...
+      case 'dart':
+        throw ArgumentError.value(uri, 'uri', 'dart: scheme is not supported.');
+      case 'asset':
+        var parts = uri.pathSegments;
+        assetId = AssetId(parts[0], path.joinAll(parts.skip(1)));
+        break;
+      default:
+        assetId = _resolve(null, uri.toString());
+        if (assetId == null) {
+          log.severe('Unable to resolve asset ID for "$uri"');
```
Comment (jakemac53) [inline build_resolvers/lib/src/resolver.dart]: Should we throw here also then?

Comment (kevmoo) [reply build_resolvers/lib/src/resolver.dart]: 🤷‍♂️

**Comment by jakemac53** [comment]
It looks like we should actually just be returning null if we can't handle the URI, can you try changing it to do that @kevmoo ?

**Comment by kevmoo** [comment]
> can you try changing it to do that

Sure. Should we at least warn? Or nothing?

**Comment by jakemac53** [comment]
> Sure. Should we at least warn? Or nothing?

I think nothing, my guess is the analyzer will end up throwing some sort of exception if nothing handles the uri, so we might want to handle that but in a different place.

**Comment by kevmoo** [comment]
PTAL

**Pull request merged**